### PR TITLE
Add gql yoga plugin to properly set cookies in response headers

### PR DESCRIPTION
**Pull request by nathanchica**
This pull request introduces a new `contextHeadersPlugin` to streamline the transfer of headers from the GraphQL context to HTTP responses. It also includes updates to integrate this plugin into the GraphQL Yoga server configurations and adds comprehensive tests to ensure its functionality. Additionally, it refines cookie settings for better development and production handling.

### Plugin Integration and Configuration Changes:
* Added `contextHeadersPlugin` to transfer headers from `context.response` to HTTP responses using a `WeakMap` for request-context mapping (`apps/api/src/utils/plugins.ts`).
* Integrated `contextHeadersPlugin` into the GraphQL Yoga server configurations in `apps/api/api/graphql.ts` and `apps/api/src/main.ts`. [[1]](diffhunk://#diff-e918435f61bec63b508ff38ba920c79a030704a6790dd7fbcde5b1e0b15d5ac1R7) [[2]](diffhunk://#diff-e918435f61bec63b508ff38ba920c79a030704a6790dd7fbcde5b1e0b15d5ac1L26-R28) [[3]](diffhunk://#diff-32c5294dd7ca131397ef57a87c3e0cf38ea5c2e4ca0061c602e73f7f00514c1aR8) [[4]](diffhunk://#diff-32c5294dd7ca131397ef57a87c3e0cf38ea5c2e4ca0061c602e73f7f00514c1aL16-R18)

### Tests for `contextHeadersPlugin`:
* Added extensive unit tests for `contextHeadersPlugin` to verify header transfer, edge cases, and error handling in `apps/api/src/utils/__tests__/plugins.test.ts`.

### Cookie Settings Update:
* Refined `COOKIE_OPTIONS` to dynamically set `sameSite` based on the environment, enhancing cross-origin cookie handling in development and security in production (`apps/api/src/schema/user/index.ts`).

### CODE REVIEW

File: apps/api/src/schema/user/index.ts [hunk 1]
```diff
@@ -39,9 +39,13 @@ const GoogleOAuthLoginMutationSchema = z.object({
...
     secure: env.NODE_ENV === 'production',
-    sameSite: 'lax' as const,
     maxAge: env.REFRESH_TOKEN_EXPIRES_IN_DAYS * MILLISECONDS_IN_DAY,
-    path: '/'
+    path: '/',
+    // In development, omit sameSite to allow cross-origin cookies
+    // In production, use 'lax' for security
+    /* v8 ignore start */
+    ...(env.NODE_ENV === 'production' && { sameSite: 'lax' as const })
+    /* v8 ignore stop */
```
[the remark this answers is not in the document]
Comment (nathanchica) [reply apps/api/src/schema/user/index.ts]: converting to a ternary instead of &&
